pkgs/by-name/di/dingtalk/package.nix

```
@@ -0,0 +1,192 @@
+  gtk2-x11,
+}:
+let
+  libraries = [
+    alsa-lib
+    apr
+    aprutil
+    at-spi2-atk
+    at-spi2-core
+    cups
+    gtk3
+    libpulseaudio
+    mesa
+    nspr
+    nss
+    qt5.qtbase
+    qt5.qtmultimedia
+    qt5.qtsvg
+    qt5.qtx11extras
+    xorg.libX11
+    xorg.libxcb
+    xorg.libXcomposite
+    xorg.libXcursor
+    xorg.libXdamage
+    xorg.libXext
+    xorg.libXfixes
+    xorg.libXi
+    xorg.libXinerama
+    xorg.libXrandr
+    xorg.libXrender
+    xorg.libXScrnSaver
+    xorg.libXtst
+    glib
+    gdk-pixbuf
+    freetype
+    fontconfig
+    libdrm
+    libxkbcommon
+    libGLU
+    libglvnd
+    udev
+    libuuid
+    libappindicator-gtk3
+    libdbusmenu-gtk3
+    gtk2-x11
+    gnome2.gtkglext
+    openssl_1_1
+  ];
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "dingtalk";
+  version = "7.6.45.5062501";
+  src =
+    let
+
+      selectSystem =
+        attrs:
+        attrs.${stdenv.hostPlatform.system}
+          or (throw "dingtalk: ${stdenv.hostPlatform.system} is not supported");
+      arch = selectSystem {
+        x86_64-linux = "amd64";
+        aarch64-linux = "arm64";
+      };
+    in
+    fetchurl {
+      url = "https://dtapp-pub.dingtalk.com/dingtalk-desktop/xc_dingtalk_update/linux_deb/Release/com.alibabainc.dingtalk_${finalAttrs.version}_${arch}.deb";
+      hash = selectSystem {
+        x86_64-linux = "sha256-49pmBducAxUy7ZqQg+tdDquFeR+lJczpcF8m+VvhIo8=";
+        aarch64-linux = "sha256-ebIKFfJ8s3w0yGXapTWYtPSz8QgLsN6ujj2k9mQFR2k=";
+      };
+    };
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    makeWrapper
+    qt5.wrapQtAppsHook
+    copyDesktopItems
+    dpkg
+  ];
+
+  buildInputs = libraries;
+  dontWrapQtApps = true;
+
+  installPhase = ''
+    runHook preInstall
+
+    # Extract the package
+    dpkg -x $src .
+
+    # Move version file
+    mv opt/apps/com.alibabainc.dingtalk/files/version version
+
+    # Move main files
+    mv opt/apps/com.alibabainc.dingtalk/files/*-Release.* release
+
+    # Remove unnecessary files
+    rm -f release/{*.a,*.la,*.prl,dingtalk_crash_report,dingtalk_updater,libapr*,libcrypto.so.*,libcurl.so.*}
+    rm -f release/{libdouble-conversion.so.*,libEGL*,libfontconfig*,libfreetype*,libfribidi*,libgdk*,libGLES*}
+    rm -f release/{libgtk*,libgtk-x11-2.0.so.*,libharfbuzz*,libicu*,libidn2*,libjpeg*,libm.so.*,libnghttp2*}
+    rm -f release/{libpango-1.0.*,libpangocairo-1.0.*,libpangoft2-1.0.*,libpcre2*,libpng*,libpsl*,libQt5*,libssh2*}
+    rm -f release/{libssl.*,libstdc++.so.6,libstdc++*,libunistring*,libvk*,libvulkan*,libxcb*,libz*}
+    rm -rf release/{engines-1_1,imageformats,platform*,swiftshader,xcbglintegrations}
+    rm -rf release/Resources/{i18n/tool/*.exe,qss/mac}
+
+    # Install files
+    install -Dm644 version $out/version
+    # Use libexec directory for application binaries as it's more suitable than lib
+    mv release $out/libexec
+    mkdir -p $out/bin
+    ln -s $out/libexec/com.alibabainc.dingtalk $out/bin/dingtalk
```

```suggestion
    ln -s $out/libexec/dingtalk/com.alibabainc.dingtalk $out/bin/dingtalk
```
use $out/libexec/dingtalk